scripts/bottleneck_detector.py

```
@@ -0,0 +1,583 @@
+                    description=f"Agents {overloaded_agents} have high workloads "
+                              f"(>{backlog_threshold} tasks)",
+                    affected_steps=[],
+                    affected_agents=overloaded_agents,
+                    timestamp=time.time(),
+                    resolution_suggestion="Consider adding more agents or redistributing workload"
+                )
+                alerts.append(alert)
+                
+        return alerts
+        
+    def detect_resource_contention(self, resource_threshold: float = 80.0) -> List[BottleneckAlert]:
```

Ruff (0.14.4)

Unused method argument: `resource_threshold`

(ARG002)